test/lib/jdk/test/lib/util/ForceGC.java

```diff
@@ -34,15 +34,18 @@
 public class ForceGC {
     private final CountDownLatch cleanerInvoked = new CountDownLatch(1);
     private final Cleaner cleaner = Cleaner.create();
+    private Object o;
+
     public ForceGC() {
-        cleaner.register(new Object(), () -> cleanerInvoked.countDown());
+        this.o = new Object();
+        cleaner.register(o, () -> cleanerInvoked.countDown());
     }
 
-    private void doit() {
+    private void doit(int iter) {
         try {
             for (int i = 0; i < 10; i++) {
                 System.gc();
-                System.out.println("gc " + i);
+                System.out.format("doit %d: gc %d%n", iter, i);
                 if (cleanerInvoked.await(1L, TimeUnit.SECONDS)) {
```

Comment:
If the object that the test waits for garbage collected takes more GC cycles than the object registered in ForceGC's cleaner, this will return immediately.   If you are concerned, maybe if the count down latch already becomes zero, it should call System.sleep before the next System.gc().

Reply:
I am a little concerned.  But I think it makes more sense to sleep in await(), between calls of doit().  I'll push an update.